compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -898,34 +1245,53 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
             }
         }
 
-        ty::FnDef(def_id, args) => {
-            ty = Ty::new_fn_def(tcx, *def_id, transform_args(tcx, args, options));
+        ty::FnDef(def_id, ..) => {
```

Comment:
Needs explanation explaining why it's being done.

Reply:
This is also related to the information I provided above, but in a different manner. Since function items, closures, and Fn trait objects can be used as arguments to functions that receive any of these types as parameters, we need to find and use the least common denominator to represent them when encoding and assigning labels to a function or method, and later use that same representation at the call sites, so the labels match. For these types, these are their return and parameter types (but let me know if there is any other possible representation that would meet the constraints I explained above).